src/status_im/contexts/onboarding/events.cljs

```diff
@@ -184,27 +183,41 @@
  (fn [{:keys [db]} [auth-method]]
    {:db (assoc db :auth-method auth-method)}))
 
-(rf/defn onboarding-new-account-finalize-setup
-  {:events [:onboarding/finalize-setup]}
-  [{:keys [db]}]
-  (let [masked-password    (get-in db [:onboarding/profile :password])
-        key-uid            (get-in db [:profile/profile :key-uid])
-        syncing?           (get-in db [:onboarding/profile :syncing?])
-        auth-method        (get-in db [:onboarding/profile :auth-method])
-        biometric-enabled? (= auth-method constants/auth-method-biometric)]
-    (cond-> {:db (assoc db :onboarding/generated-keys? true)}
-      biometric-enabled?
-      (assoc :keychain/save-password-and-auth-method
-             {:key-uid         key-uid
-              :masked-password (if syncing?
-                                 masked-password
-                                 (security/hash-masked-password masked-password))
-              :on-success      (fn []
-                                 (rf/dispatch [:onboarding/set-auth-method auth-method])
-                                 (when syncing?
-                                   (rf/dispatch
-                                    [:onboarding/navigate-to-enable-notifications-from-syncing])))
-              :on-error        #(log/error "failed to save biometrics"
-                                           {:key-uid key-uid
-                                            :error   %})}))))
+(def ^:const temp-display-name
```

Comment:
This area of code is new to me and have to say, I felt quite confused trying to understand the difference between `temporary-display-name` and `default-profile-name` and how they are used.
First few minutes I was reading `set-reset-display-name` as `(set-reset)-display-name` instead of `(set)-reset-display-name` and thought that it is some kind of ambiguous event that decides what to do (set or reset) depending on the argument.
Then I didn't get that `temporary-name` and `default-name` are different concepts.
Probably I'm grasping too slow but overall that took me around 15 minutes :)

So I'd suggest some renaming for code readability which I believe would make me understand things faster: 

---

**1.** `:onboarding/set-reset-display-name` -> `:onboarding/use-temporary-display-name`
```clojure
(rf/reg-event-fx
 :onboarding/use-temporary-display-name
 (fn [{:keys [db]} [use-temporary-name?]]
   (let [initial-profile-data {:current-name-is-temporary? use-temporary-name? 
                               :display-name            (if use-temporary-name? temp-display-name "")}]
     {:db (assoc db :onboarding/profile initial-profile-data)})))
```

**2.** `get-default-display-name` -> `create-display-name-from-profile-key`
```clojure
(defn create-display-name-from-profile-key
  [profile]
  (-> profile :compressed-key (string/split #"zQ3") second (subs 0 constants/profile-name-max-length)))
```

**3.** `:profile/set-default-profile-name` -> `:profile/set-profile-name-generated-from-key`

---

With such renaming the `when` condition in `:onboarding/finalize-setup` will look like:
```clojure
..
:fx [(when current-name-is-temporary?
             [:dispatch [:profile/set-profile-name-generated-from-key profile]])
             ...
```

And calls that I found confusing:
`(rf/dispatch [:onboarding/set-reset-display-name false])`
will look like this:
`(rf/dispatch [:onboarding/use-temporary-display-name false])`

@ulisesmac, wdyt?

Reply:
@vkjr Thank you so much for your feedback on the naming. I agree it's a bit confusing. :+1: 

I applied most of your suggestions, I'll give my explanation:

> 1. :onboarding/set-reset-display-name -> :onboarding/use-temporary-display-name

Agree! this is a better name!

> :display-name-is-temporary? 

This name adding `is` sounds non-Clojure friendly to me, sounds a bit like a Java predicate (because of `is`), so I used `temporary-display-name?` instead.

> get-default-display-name -> create-display-name-from-profile-key

Sounds a bit more verbose than needed to me, since from the signature it's clear that we receive a `profile`, also I'd omit the `create-` prefix.

I used `display-name-from-compressed-key` wdyt?

> :profile/set-default-profile-name -> :profile/set-profile-name-generated-from-key

I think in this case this renaming is not needed, since this event is actually setting the **default** profile name, It's also not needed to expose all details in the event name, if we desire to check what is that "default profile name", the implementation is very straight-forward:

```clojure
(let [default-display-name (display-name-from-compressed-key profile)]
  {:db (assoc-in db [:profile/profile :display-name] default-display-name)
   :fx [[:json-rpc/call
         [{:method "wakuext_setDisplayName"
           :params [default-display-name]}]]]})
```

Reply:
@ulisesmac, agree on all points, thank you!